### PR TITLE
Rework `debug` docs and add small section on troubleshooting

### PR DESCRIPTION
**Pull request by briandealwis**

### CODE REVIEW

File: docs/content/en/docs/workflows/debug.md [hunk 1]
```diff
@@ -57,33 +75,53 @@ specified in [Skaffold's global configuration]({{< relref "/docs/design/global-c
...
 Debugging is currently supported for:
-  - Go (runtime ID: `go`)
-  - NodeJS (runtime ID: `nodejs`)
-  - Java and JVM languages (runtime ID: `jvm`)
-  - Python (runtime ID: `python`)
-  - .NET Core (runtime ID: `netcore`)
+  - Go (runtime ID: `go`) using [Delve](https://github.com/go-delve/delve)
+  - NodeJS (runtime ID: `nodejs`) using the NodeJS Inspector (Chrome DevTools)
+  - Java and JVM languages (runtime ID: `jvm`) using JDWP
+  - Python (runtime ID: `python`) using `debugpy` (Debug Adapter Protocol) or `pydevd`
```
Comment (aaron-prindle) [inline docs/content/en/docs/workflows/debug.md:84]: are all Pythons version supported?

File: docs/content/en/docs/workflows/debug.md [hunk 2]
```diff
@@ -127,27 +183,54 @@ only invokes the real <tt>node</tt> with <tt>--inspect</tt> if running an applic
...
+- the presence of a standard .NET environment variable:
+  `ASPNETCORE_URLS`, `DOTNET_RUNNING_IN_CONTAINER`,
+  `DOTNET_SYSTEM_GLOBALIZATION_INVARIANT`, or
+- the container command-line invokes the [dotnet](https://github.com/dotnet/sdk) cli
 
-- Identified as being dotnet-based by having an entrypoint using [dotnet](https://github.com/dotnet/sdk) cli
-  or one of the following environment variables `ASPNETCORE_URLS`, `DOTNET_RUNNING_IN_CONTAINER`,
-  `DOTNET_SYSTEM_GLOBALIZATION_INVARIANT`.
-- Built with the `--configuration Debug` options to disable optimizations.
+Furthermore, your app must be built with the `--configuration Debug` options to disable optimizations.
```
Comment (aaron-prindle) [inline docs/content/en/docs/workflows/debug.md:233]: is the `Debug` (vs `debug`) uppercase correct here?  Verifying it works as suggested.

**Review by aaron-prindle — APPROVED** [review]
LGTM, with some small nit comments/questions

File: docs/content/en/docs/workflows/debug.md (hunk 1 above)
Comment (aaron-prindle) [reply docs/content/en/docs/workflows/debug.md:84]: nit: if might make sense for supported versions for each runtime (if there is such limitation) be put here or in the langauge specific sections

File: docs/content/en/docs/workflows/debug.md (hunk 2 above)
Comment (briandealwis) [reply docs/content/en/docs/workflows/debug.md:233]: It worked for me — this contribution is community provided, and it's difficult to test.

File: docs/content/en/docs/workflows/debug.md (hunk 1 above)
Comment (briandealwis) [reply docs/content/en/docs/workflows/debug.md]: I moved the runtimes lists to the top and added versions where it makes sense.

**Pull request merged**